code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -34,94 +34,501 @@
+
+		if(SLIME_ACTIVATE_MAJOR)
+			to_chat(user, "<span class='notice'>You activate [src], and it releases regenerative chemicals!</span>")
+			user.reagents.add_reagent("tricordrazine",10)
+			return 600
+
 /obj/item/slime_extract/darkpurple
 	name = "dark purple slime extract"
 	icon_state = "dark purple slime extract"
 
+/obj/item/slime_extract/darkpurple/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			var/obj/item/stack/sheet/mineral/plasma/O = new(null, 1)
+			if(!user.put_in_active_hand(O))
+				O.forceMove(user.drop_location())
+			playsound(user, 'sound/effects/splat.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] spits out [O]!</span>", "<span class='notice'>You spit out [O]!</span>")
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			var/turf/open/T = get_turf(user)
+			if(istype(T))
+				T.atmos_spawn_air("plasma=20")
+			to_chat(user, "<span class='warning'>You activate [src], and a cloud of plasma bursts out of your skin!</span>")
+			return 900
+
 /obj/item/slime_extract/orange
 	name = "orange slime extract"
 	icon_state = "orange slime extract"
 
+/obj/item/slime_extract/orange/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			to_chat(user, "<span class='notice'>You activate [src]. You start feeling hot!</span>")
+			user.reagents.add_reagent("capsaicin",10)
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			user.reagents.add_reagent("phosphorus",5)//
+			user.reagents.add_reagent("potassium",5) // = smoke, along with any reagents inside mr. slime
+			user.reagents.add_reagent("sugar",5)     //
+			to_chat(user, "<span class='warning'>You activate [src], and a cloud of smoke bursts out of your skin!</span>")
+			return 450
+
 /obj/item/slime_extract/yellow
 	name = "yellow slime extract"
 	icon_state = "yellow slime extract"
 
+/obj/item/slime_extract/yellow/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			if(species.glow_intensity != LUMINESCENT_DEFAULT_GLOW)
+				to_chat(user, "<span class='warning'>Your glow is already enhanced!</span>")
+				return
+			species.update_glow(user, 5)
+			addtimer(CALLBACK(species, /datum/species/jelly/luminescent.proc/update_glow, user, LUMINESCENT_DEFAULT_GLOW), 600)
+			to_chat(user, "<span class='notice'>You start glowing brighter.</span>")
+
+		if(SLIME_ACTIVATE_MAJOR)
+			user.visible_message("<span class='warning'>[user]'s skin starts flashing intermittently...</span>", "<span class='warning'>Your skin starts flashing intermittently...</span>")
+			if(do_after(user, 25, target = user))
+				empulse(user, 1, 2)
+				user.visible_message("<span class='warning'>[user]'s skin flashes!</span>", "<span class='warning'>Your skin flashes as you emit an EMP pulse!</span>")
```

Comment:
EMP pulse = electromagnetic pulse pulse

Reply:
Outright lethally strong versus borgs & people with EMP vunerable auguments and organs.

This is never getting ported to downstreams with IPC's

Reply:
It has 2.5 seconds of warning while you stand still, borgs can roll away. Also, you can do a pulse twice as strong instantly by injecting the core with plasma.

Reply:
Stun the borg with a flash, chain into EMP at close range to reset the stun and batter until dead with the additional damage taken from the EMP. Nobody enjoys fighting changelings with a EMP scream for the same reasons.

Reply:
Stun with a flash, chain into extinguisher x3, flash again, repeat. The flash is the key factor here.

Reply:
If you manage to flash the borg, you have already won. Flash, beat for the minimum stun duration, flash again, continue beating until borg is dead.